patches/chromium/feat_add_streaming-protocol_registry_to_multibuffer_data_source.patch

```diff
@@ -0,0 +1,68 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Paul Frazee <pfrazee@gmail.com>
+Date: Tue, 28 Apr 2020 17:32:18 -0500
+Subject: feat: add streaming-protocol registry to multibuffer_data_source
+
+blink::WebMediaPlayerImpl - which provides the <video> and <audio> behaviors - needs to know
+whether a data source will stream or fully buffer the response. It determines this behavior
+with MultibufferDataSource::AssumeFullyBuffered() which has http/s hardwired. An incorrect
+determination will cause the video/audio to fail playing.
+
+This patch adds a list of "streaming protocols" to the MultibufferDataSource in order to allow
+other protocols to register their streaming behavior. MultibufferDataSource::AssumeFullyBuffered()
+then refers to the list so that it can correctly determine the data source's settings.
+
+diff --git a/media/blink/multibuffer_data_source.cc b/media/blink/multibuffer_data_source.cc
+index 0f6ae1fb8b4ff9f24ce3f407b7359e016fc6de5f..3f7469376b70b0fb9ab623cc5b415a69a5640cf0 100644
+--- a/media/blink/multibuffer_data_source.cc
++++ b/media/blink/multibuffer_data_source.cc
+@@ -65,10 +65,20 @@ const int kUpdateBufferSizeFrequency = 32;
+ // How long to we delay a seek after a read?
+ constexpr base::TimeDelta kSeekDelay = base::TimeDelta::FromMilliseconds(20);
+ 
++// Schemes that can stream their responses.
++std::vector<std::string> streaming_schemes = {
++    url::kHttpsScheme,
++    url::kHttpScheme
++};
```

Comment:
Use `base::NoDestructor` here (see `GetSchemeRegistryWithoutLocking` in `//url/url_util.cc` for an example.)

Reply:
Went ahead and copied their entire approach of putting the schemes in a struct and applying locking schemes